department-of-veterans-affairs/abd-vro
# MCP-1985: audit event infrastructure

## What was the problem?

We need a mechanism to capture the lifecycles of claims, automated claims, etc and persiste them as described in (https://amida.atlassian.net/browse/MCP-1983)

Associated tickets or Slack threads:

- [MCP-1985](https://amida.atlassian.net/browse/MCP-1985)

## How does this fix it?[^1]

This PR addresses the infrastructure part of the ticket. The mechanism is introduced to log events as part of camel routes. The persistence, queues, and notification mechanisms will be addressed in the remaining tasks.

## How to test this PR
- This is not yet testable

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
